tests/org.eclipse.jface.tests/src/org/eclipse/jface/tests/viewers/AbstractTreeViewerTest.java

```diff
@@ -164,6 +164,144 @@ public void testExpandToLevel() {
+				fTreeViewer.getExpandedState(trivialPathRoot));
+		assertTrue("The first child of the trivial path was not auto-expanded",
+				fTreeViewer.getExpandedState(trivialPathRoot.getFirstChild()));
+		assertTrue("The second child of the trivial path was not auto-expanded",
+				fTreeViewer.getExpandedState(trivialPathRoot.getFirstChild().getFirstChild()));
+		assertFalse("Trivial path is expanded further than specified depth ",
+				fTreeViewer.getExpandedState(trivialPathRoot.getFirstChild().getFirstChild().getFirstChild()));
+	}
+
+	@Test
+	public void testAutoExpandTrivialPathFromRoot() {
+		TestElement modelRoot = TestElement.createModel(5, 5);
+		TestElement trivialPathRoot = modelRoot;
+		fViewer.setInput(modelRoot);
+
+		fTreeViewer.setAutoExpandOnSingleChildLevels(2);
+		fTreeViewer.setExpandedStateWithAutoExpandOnSingleChild(trivialPathRoot, true);
+
+		assertFalse("The first child of the trivial path was auto-expanded",
+				fTreeViewer.getExpandedState(trivialPathRoot.getFirstChild()));
+	}
+
+	@Test
+	public void testAutoExpandTrivialPathFromRootWithSensibleTrivialPath() {
+		TestElement modelRoot = TestElement.createModel(5, 1);
+		TestElement trivialPathRoot = modelRoot;
+		fViewer.setInput(modelRoot);
+
+		fTreeViewer.setAutoExpandOnSingleChildLevels(2);
+		fTreeViewer.setExpandedStateWithAutoExpandOnSingleChild(trivialPathRoot, true);
+
+		assertTrue("The first child of the trivial path was auto-expanded",
+				fTreeViewer.getExpandedState(trivialPathRoot.getFirstChild()));
+	}
+
+	@Test
+	public void testAutoExpandTrivialPathSmallerThanAutoExpandDepth() {
+		TestElement modelRoot = TestElement.createModel(2, 1);
+		TestElement trivialPathRoot = modelRoot.getFirstChild();
+		fViewer.setInput(modelRoot);
+
+		fTreeViewer.setAutoExpandOnSingleChildLevels(10);
+		fTreeViewer.setExpandedStateWithAutoExpandOnSingleChild(trivialPathRoot, true);
+
+		assertTrue("The expanded widget child is not expanded", fTreeViewer.getExpandedState(trivialPathRoot));
+		assertFalse("The first child of the trivial path was auto-expanded although it contains zero children",
+				fTreeViewer.getExpandedState(trivialPathRoot.getFirstChild()));
+	}
+
+	@Test
+	public void testSetAutoExpandOnSingleChildLevels() {
+		fTreeViewer.setAutoExpandOnSingleChildLevels(AbstractTreeViewer.NO_EXPAND);
+		assertEquals("Setting an auto-expansion level of NO_EXPAND works",
+				fTreeViewer.getAutoExpandOnSingleChildLevels(), AbstractTreeViewer.NO_EXPAND);
+
+		fTreeViewer.setAutoExpandOnSingleChildLevels(4);
+		assertEquals("Setting a non-trivial auto-expansion level works", fTreeViewer.getAutoExpandOnSingleChildLevels(),
+				4);
+
+		fTreeViewer.setAutoExpandOnSingleChildLevels(AbstractTreeViewer.NO_EXPAND);
+		assertEquals("Setting an auto-expansion level of NO_EXPAND works",
+				fTreeViewer.getAutoExpandOnSingleChildLevels(), AbstractTreeViewer.NO_EXPAND);
+	}
+
+	@Test
+	public void testAutoExpandTrivialPathManualDisable() {
```

Comment:
```suggestion
	public void testAutoExpandOnSingleChildManualDisable() {
```

Reply:
Addressed in https://github.com/eclipse-platform/eclipse.platform.ui/commit/fb256ed9423ce9f552aa278fa3893d833ccc0e63